tests/backup/backup_resiliency_test.go

```diff
@@ -1181,25 +1188,16 @@ var _ = Describe("{RebootNodesWhenBackupsAreInProgress}", func() {
 				},
 			})
 			dash.VerifyFatal(err, nil, fmt.Sprintf("Rebooting worker node %v", listOfWorkerNodes[0].Name))
-			log.InfoD("Switching cluster context back to source cluster")
-			err = SetSourceKubeConfig()
-			log.FailOnError(err, "Switching context to source cluster")
-
 		})
 		Step("Check if backup is successful after one worker node on application cluster is rebooted", func() {
 			log.InfoD("Check if backup is successful after one worker node on application cluster is rebooted")
-			ctx, err := backup.GetAdminCtxFromSecret()
-			log.FailOnError(err, "Fetching px-central-admin ctx")
 			for _, backupName := range backupNames {
-				err := backupSuccessCheck(backupName, orgID, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second, ctx)
-				dash.VerifyFatal(err, nil, "Inspecting backup success for - "+backupName)
+				err := backupSuccessCheckWithValidation(ctx, backupName, appContextsToBackupMap[backupName], orgID, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of success and Validation of the backup [%s]", backupName))
 			}
 		})
 		Step("Check if the rebooted node on application cluster is up now", func() {
 			log.InfoD("Check if the rebooted node on application cluster is up now")
-			log.InfoD("Switching cluster context to destination cluster")
-			err := SetDestinationKubeConfig()
```

Comment:
Why are we removing  this?

Reply:
Are you sure that it's doing the operator on the destination cluster ? If no then you are changing functionality which is not correct.

Reply:
yes, I am sure of it. The only reason we were switching was to get `backup.GetAdminCtxFromSecret()`, which I moved to `JustBeforeEach` so that we don' need to get it over and over again (old flawed method). So, context switches are no longer needed, which means we're almost always on the destination cluster and almost never need to be on source.

In conclusion, no, I'm not changing functionality incorrectly.